pkg/plugin/work-in-progress/event_handler.go

```diff
@@ -178,3 +146,89 @@ func (gh *GitHubWIPPRHandler) checkTitleAndSetStatus(log log.Logger, pullRequest
 	return statusService.Success(ReadyForReviewMessage, ReadyForReviewDetailsPageName)
 
 }
+
+func (gh *GitHubWIPPRHandler) updateTitleAndResetStatus(log log.Logger, pullRequest *gogh.PullRequest) error {
+	change := gh.getRepositoryChange(pullRequest)
+	statusService := gh.getStatusService(log, change)
+	labels := gh.getPullRequestLabels(log, change, pullRequest)
+	configuration := LoadConfiguration(log, change)
+
+	labelExists := gh.hasWorkInProgressLabel(labels, configuration.Label)
+	if !labelExists {
+		if prefix, ok := gh.IsWorkInProgress(*pullRequest.Title, configuration); ok {
+			*pullRequest.Title = strings.TrimPrefix(*pullRequest.Title, prefix)
+			err := gh.Client.EditPullRequest(change, *pullRequest.Number, pullRequest)
+			if err != nil {
+				log.Warnf("failed to update PR title [%q]. cause: %s", *pullRequest, err)
+			}
+		}
+		return statusService.Success(ReadyForReviewMessage, ReadyForReviewDetailsPageName)
```

Comment:
if the label was removed, but the edit of the PR [failed](https://github.com/arquillian/ike-prow-plugins/pull/189/files#diff-11865c15f9f479c8ca04795d06b465acR161) should we change the status to success as well? then it looks a bit inconsistent - the PR contains the prefix, but the status is green...
maybe, instead of logging the error https://github.com/arquillian/ike-prow-plugins/pull/189/files#diff-11865c15f9f479c8ca04795d06b465acR162 we should return it